flowzone.yml

```diff
@@ -3120,15 +3120,30 @@ jobs:
 
       - name: Generate shared outputs
         id: shared
+        env:
+          SECRETS_CONTEXT: ${{ toJson(secrets) }}
+          VARS_CONTEXT: ${{ toJson(vars) }}
         run: |
-          stack="$(echo '${{ inputs.cloudformation_templates }}' | jq -r '.stacks[] | select(.name=="${{ matrix.stack }}")')"
+          set -a
+
+          cleanup() { rm -f .env; }
+          trap cleanup EXIT
+
+          to_envs() { jq -r "to_entries[] | \"\(.key)="\'"\(.value)"\'"\""; }
 
+          stack="$(echo '${{ inputs.cloudformation_templates }}' | jq -r '.stacks[] | select(.name=="${{ matrix.stack }}")')"
           template_file="$(echo "${stack}" | jq -rc .template)"
           tags="$(echo "${stack}" | jq -rc .tags[] | paste -sd' ' -) github_pull_request=${{ github.event.pull_request.number }} github_sha=${{ github.event.pull_request.head.sha || github.event.head_commit.id }}"
           params="$(echo "${stack}" | jq -rc .params[] | paste -sd' ' -)"
           kvparams="$(echo "${stack}" | jq -rc .params | jq -r 'map(split("=") as [$ParameterKey, $ParameterValue] | {$ParameterKey, $ParameterValue})[] | "ParameterKey=" + .ParameterKey + ",ParameterValue=" + .ParameterValue' | paste -sd' ' -)"
           caps="$(echo "${stack}" | jq -rc .capabilities[] | paste -sd' ' -)"
 
+          echo "${SECRETS_CONTEXT}" | to_envs > .env
+          echo "${VARS_CONTEXT}" | to_envs >> .env
+          source .env
+          params="$(echo "${params}" | envsubst)"
```

Comment:
Honestly I would try to clean up here as well, really as soon as possible after sourcing the file.